src/Tasks/Microsoft.ManagedLanguages.targets

```diff
@@ -0,0 +1,64 @@
+﻿<!--
+***********************************************************************************************
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+This file defines common build logic for all managed languaged: C#, VisualBasic, F#
+
+Copyright (C) Microsoft Corporation. All rights reserved.
+***********************************************************************************************
+-->
+
+<Project>
+   <!-- 
+        We are doing a cross-targeting build if there is a non-empty list of target frameworks specified
+        and there is no current target framework being built individually. In that case, a crosstargeting
+        project file like Microsoft.<language>.CrossTargeting.targets gets imported.
+   -->
+   <PropertyGroup Condition="'$(TargetFrameworks)' != '' and '$(TargetFramework)' == ''">
+      <IsCrossTargetingBuild>true</IsCrossTargetingBuild>
+   </PropertyGroup>
+
+   <PropertyGroup>
+      <InnerBuildProperty>TargetFramework</InnerBuildProperty>
+      <InnerBuildPropertyValues>TargetFrameworks</InnerBuildPropertyValues>
+   </PropertyGroup>
+
+   <PropertyGroup Condition="'$(IsGraphBuild)' == 'true'">
```

Comment:
Should we rename to `MSBuildIsGraphBuild` and make it a reserved property, so you can't set it manually?

Reply:
@AndyGerlicher suggests not constraining it: https://github.com/Microsoft/msbuild/pull/4143#issuecomment-461198290

Making it reserved avoids an entire class of problems that can result from people manually setting this. On the other hand, power users might enjoy shooting themselves in the foot :)